test-utils/src/main/kotlin/com/google/devtools/ksp/processor/MapClassSignatureProcessor.kt

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2020 Google LLC
+ * Copyright 2010-2020 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.devtools.ksp.processor
+
+import com.google.devtools.ksp.KspExperimental
+import com.google.devtools.ksp.getClassDeclarationByName
+import com.google.devtools.ksp.processing.Resolver
+import com.google.devtools.ksp.symbol.KSAnnotated
+
+@KspExperimental
+class MapClassSignatureProcessor : AbstractTestProcessor() {
```

Comment:
We have `MapSignatureProcessor` where you can simply add an enum class there and add it to the visit list.

Reply:
That processor also tests methods, and the results don't match between two tests, so I had to create a new one.

Reply:
Did you mean the result don't match between compiler-plugin and kotlin-analysis-api tests?

Reply:
Yes

Reply:
some difference between the 2 is acceptable depending on the case, it's better to just split test data rather than writing a seemingly unnecessary new test. IIRC there is some difference around enum member functions, if that's the case, it should be ok to just split test data.

Reply:
Ok, I did split the test data, please check